ballerina/http_client_endpoint.bal

```diff
@@ -677,17 +677,41 @@ isolated function createResponseError(int statusCode, string reasonPhrase, map<s
     }
 }
 
-isolated function createStatusCodeResponseBindingError(boolean generalError, int statusCode, string reasonPhrase,
-        map<string[]> headers, anydata body = ()) returns ClientError {
-    if generalError {
-        return error StatusCodeResponseBindingError(reasonPhrase, statusCode = statusCode, headers = headers, body = body);
-    }
+isolated function createStatusCodeResponseBindingError(int statusCode, string reasonPhrase, map<string[]> headers,
+        anydata body = ()) returns ClientError {
     if 100 <= statusCode && statusCode <= 399 {
-        return error StatusCodeBindingSuccessError(reasonPhrase, statusCode = statusCode, headers = headers, body = body);
+        return error StatusCodeResponseBindingError(reasonPhrase, statusCode = statusCode, headers = headers, body = body, fromDefaultStatusCodeMapping = false);
     } else if 400 <= statusCode && statusCode <= 499 {
-        return error StatusCodeBindingClientRequestError(reasonPhrase, statusCode = statusCode, headers = headers, body = body);
+        return error StatusCodeBindingClientRequestError(reasonPhrase, statusCode = statusCode, headers = headers, body = body, fromDefaultStatusCodeMapping = false);
     } else {
-        return error StatusCodeBindingRemoteServerError(reasonPhrase, statusCode = statusCode, headers = headers, body = body);
+        return error StatusCodeBindingRemoteServerError(reasonPhrase, statusCode = statusCode, headers = headers, body = body, fromDefaultStatusCodeMapping = false);
+    }
+}
+
+isolated function createStatusCodeResponseDataBindingError("header"|"mediaType"|"payload"|"generic" errorType, boolean fromDefaultStatusCodeMapping,
```

Comment:
Can't we create an enum or a type-def for error type and use it here ?

```ballerina
type DataBindingErrorType "header"|"mediaType"|"payload"|"generic";
```

or

```ballerina

enum DataBindingErrorType {
    HEADER, MEDIA_TYPE, PAYLOAD, GENERIC
}
```

Reply:
This is used for internal purpose(non-public and called from java native function) and used only at this function. Thats why I have just defined it as the union of singletons

Reply:
Fixed via [3ed8b32](https://github.com/ballerina-platform/module-ballerina-http/pull/2002/commits/3ed8b32953953b57dc0074a09681ce99e755e232)